### PR TITLE
User Segment calculations

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/72649045549333/1206955287217878/f

### Description
Add user segmentation

### Steps to test this PR
The tests added should ensure the functionality works as expected.

To run manual test, one needs to set up charles and proxy/change the `/atb.js` response

### CODE REVIEW

**Comment by aitorvs** [comment]
* **#4965** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4965?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @aitorvs and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: statistics/statistics-impl/src/main/java/com/duckduckgo/app/statistics/api/StatisticsService.kt [hunk 1]
```diff
@@ -41,6 +41,7 @@ interface StatisticsService {
         @Query(ParamKey.EMAIL) email: Int?,
     ): Observable<Atb>
 
+    // ANA
```
Comment (anikiki) [inline statistics/statistics-impl/src/main/java/com/duckduckgo/app/statistics/api/StatisticsService.kt:44]: Some old comments to remove :see_no_evil: There's another one at line 53.

File: statistics/statistics-impl/src/main/java/com/duckduckgo/app/statistics/user_segments/SegmentCalculation.kt [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.experiments.api.VariantManager
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+import kotlinx.coroutines.withContext
+
+interface SegmentCalculation {
+
+    /**
+     * This method computes the user segmentation for a given [ActivityType] and u
```
Comment (anikiki) [inline statistics/statistics-impl/src/main/java/com/duckduckgo/app/statistics/user_segments/SegmentCalculation.kt:32]: This comment seems truncated: "and u" (and usage history?).

File: statistics/statistics-impl/src/main/java/com/duckduckgo/app/statistics/user_segments/SegmentCalculation.kt [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+            return false
+        }
+        if (!countAsWau(prevSetAtb, newSetAtb, cohortAtb)) {
+            return false
+        }
+
+        return newSetAtb.parseAtbWeek() == prevSetAtb.parseAtbWeek() + 1
+    }
+
+    private fun getSegmentsPrevWeek(
```
Comment (anikiki) [inline statistics/statistics-impl/src/main/java/com/duckduckgo/app/statistics/user_segments/SegmentCalculation.kt:199]: This is unused.

**Review by anikiki — APPROVED** [review]
LGTM :+1:

**Pull request merged**